src/components/forms/MemoryLimitAvailable.tsx

```diff
@@ -1,59 +1,143 @@
 import { limitToBytes } from "util/limits";
 import type { LxdProject } from "types/project";
-import { useServerEntitlements } from "util/entitlements/server";
+import type { LxdResources } from "types/resources";
+import { Icon } from "@canonical/react-components";
+import type { InstanceAndProfileFormikProps } from "./instanceAndProfileFormValues";
+import type { CreateInstanceFormValues } from "pages/instances/CreateInstance";
+import type { EditInstanceFormValues } from "pages/instances/EditInstance";
+import { useClusterMembers } from "context/useClusterMembers";
+import { CLUSTER_GROUP_PREFIX } from "util/instances";
+import ResourceLink from "components/ResourceLink";
+import { useIsClustered } from "context/useIsClustered";
+import { useResources } from "context/useResources";
 
+interface MemoryLimit {
+  range: [number, number];
+  source: "project" | "cluster-member" | "";
+}
 interface Props {
+  formik?: InstanceAndProfileFormikProps;
   project?: LxdProject;
 }
 
-const MemoryLimitAvailable: FC<Props> = ({ project }) => {
+const MemoryLimitAvailable: FC<Props> = ({ formik, project }) => {
   const notify = useNotify();
-  const { canViewResources } = useServerEntitlements();
+  const isClustered = useIsClustered();
+  const { data: clusterMembers = [] } = useClusterMembers();
+  const values = formik?.values;
+  const projectResourceMemory = limitToBytes(project?.config["limits.memory"]);
+  const currentServer = (() => {
+    const createTarget =
+      (values as CreateInstanceFormValues)?.target ?? undefined;
+    if (createTarget && !createTarget.startsWith(CLUSTER_GROUP_PREFIX)) {
+      return createTarget;
+    }
+    const location = (values as EditInstanceFormValues)?.location ?? undefined;
+    return location === "none" || !location ? undefined : location;
+  })();
 
   const {
     data: resources,
-    error,
-    isLoading,
-  } = useQuery({
-    queryKey: [queryKeys.resources],
-    queryFn: async () => fetchResources(),
-    enabled: canViewResources(),
-  });
+    error: error,
+    isLoading: isLoading,
+  } = useResources(clusterMembers, currentServer);
 
   if (isLoading) {
     return <Spinner className="u-loader" text="Loading resources..." />;
   }
-
   if (error) {
     notify.failure("Loading resources failed", error);
+    return null;
   }
 
-  const getAvailableMemory = () => {
-    if (!project?.config["limits.memory"]) {
-      return resources?.memory.total;
+  const getAvailableMemory = (resources: LxdResources[]): MemoryLimit => {
+    const sortedResourcesMemories = resources
+      .map((r) => r.memory.total)
+      .sort((a, b) => a - b);
+    const minResourceMemory = sortedResourcesMemories[0];
+    const maxResourceMemory =
+      sortedResourcesMemories[sortedResourcesMemories.length - 1];
+    if (!projectResourceMemory) {
+      return {
+        range: [minResourceMemory, maxResourceMemory],
+        source: isClustered && currentServer ? "cluster-member" : "",
```

Comment:
```suggestion
        source: currentServer ? "cluster-member" : "",
```
Current server is only set if we are clustered, so we don't need the explicit check here.